test/unit/fetchers/url_test.rb

```diff
@@ -34,7 +34,14 @@
       res.expects(:open).returns(mock_open)
       _(res).must_be_kind_of Fetchers::Url
       _(res.resolved_source).must_equal({url: 'https://chef.io/some.tar.gz', sha256: expected_shasum})
+    end
 
+    it 'handles an https URI' do
```

Comment:
I think this is a great start for a basic test!  However, I think if someone is handing you a URI, it may be because it is something more exotic than a HTTPS address. Consider adding a test with something harder to parse - perhaps a URL with an embedded password, or an FTP URI, or a file:/// URI.